blogs/13_bfcl_v3_multi_turn.html

```diff
@@ -743,73 +743,62 @@ <h4>5. API Code Validation (🧑‍💻+ 💻 )</h4>
-            to managing the flow of data and function results across multiple steps. In <i>BFCL V3 • Multi-Turn & Multi-Step</i>, our model handlers
-            are designed to handle different function call patterns—simple, parallel, and nested—across multiple
-            rounds of interaction. The distinction between these call patterns is explained in the previous section.
+          <h3 id="inference">Multi-turn Model Inference Pipeline</h3>
+          <ol>
+            <li>
+              Each entry comes with its own initial configuration, which is used to initialize the API backend state. For example, a file system backend might start with a set of pre-existing files, and a messaging API might start with a specific message inbox history. The initial configuration is loaded into the API backend at the beginning of each test entry, but it is not provided to the model directly, as the model can call exploration functions to figure out the state of the system and retrieve the necessary information.
+            </li>
+            <li>
+              At the beginning of each turn, a user query is provided to the model, along with the tools available for the task. A step_counter is also initialized to track the number of steps used in this turn; it resets to 0 at the beginning of each turn.
+            </li>
+            <li>
+              If the model makes any function calls (i.e., <code>decode_exec</code> in the model handler returns a non-empty list), the function calls are executed in the API backend (in order) and their execution results are provided back to the model. The current turn continues.
+            </li>
+            <li>
+              If the model doesn't output any valid function calls, we consider this the end of the current turn, and move on to the next turn. This could be due to either the model outputs a human-message, or the model response cannot be properly decoded into executable function calls.
+            </li>
+            <li>
+              If the model takes more than 20 steps in a turn, the turn is force-terminated and this test entry is marked as wrong; no further turns will be queried. This usually happens when the model gets stuck in a loop or is unable to make progress (e.g., keep calling <code>ls</code> in the Gorilla File System entries).
+            </li>
+          </ol>
+          <h4>Note For Multi Turn Missing Function Category</h4>
+          <p>
+            In this category, one or more functions are held out from the function list provided to the model at the beginning; they will be provided in a later turns (say, turn 3). For FC models, the added functions will just be appended to the <code>tools</code> list. But for prompting models, since we supplied all the tools at the beginning in the system prompt and it's not possible to modify the system prompt in the middle of the conversation, we will provide the held-out function definitions in the content of a user message instead.
```

Comment:
s/(say, turn 3)/(e.g., turn 3)

Reply:
And, can improve clarity of this paragraph by providing an intuitive example, similar to ones in blog 8. May not be immedicately clear to users that are not as familar with the concepts.

Reply:
s/the added functions will just be appended to the tools list/the tools list will append the added functions.

try to avoid passive tone.

Reply:
Yea good point. I was also thinking about adding an example.